src/main/java/frc/robot/commands/auton/AutonBuilder.java

```diff
@@ -0,0 +1,280 @@
+     *
+     * @param initPose The initial pose to reset swerve to.
+     * @param commands The commands to run during auton.
+     * @return The full auton sequence.
+     */
+    public SequentialCommandGroup buildAuton(Pose2d initPose, Command... commands) {
+        SequentialCommandGroup autonSequence = new SequentialCommandGroup();
+
+        autonSequence.addCommands(
+            resetSwerve(GRTUtil.mirrorAcrossField(initPose, fmsSubsystem::isRedAlliance)),
+            new ShooterFlywheelReadyCommand(shooterFlywheelSubsystem, lightBarSubsystem)
+        );
+        autonSequence.addCommands(commands);
+        autonSequence.addCommands(
+            new ShooterFlywheelStopCommand(shooterFlywheelSubsystem)
+        );
+
+        return autonSequence;
+    }
+
+    /** Starts amp side and shoots the preloaded note. */
+    public SequentialCommandGroup getTopPreloaded() {
+        return buildAuton(AutonConstants.TOP_START_POSE, shoot());
+    }
+
+    /** Starts in front of subwoofer and shoots preloaded note. */
+    public SequentialCommandGroup getMiddlePreloaded() {
+        return buildAuton(AutonConstants.MIDDLE_START_POSE, shoot());
+    }
+
+    /** Starts source side and shoots preloaded note. */
+    public SequentialCommandGroup getBottomPreloaded() {
+        return buildAuton(AutonConstants.BOTTOM_START_POSE, shoot());
+    }
+
+    /** Starts amp side. Shoots preloaded note, intakes top note, shoots note. */
+    public SequentialCommandGroup getTopTwoPiece() {
+        
+        ChoreoTrajectory startToPiece1 = Choreo.getTrajectory("C1-AmpStartToAmpNote");
+
+        return buildAuton(
+            new Pose2d(startToPiece1.getInitialPose().getTranslation(), new Rotation2d()),
+            shoot(),
+            goIntake(startToPiece1),
+            shoot()
+        );
+    }
+
+    /** Starts at subwoofer. Shoots preloaded note, intakes middle  note, shoots note. */
+    public SequentialCommandGroup getMiddleTwoPiece() {
+        
+        ChoreoTrajectory startToPiece1 = Choreo.getTrajectory("A1-SpeakerStartToSpeakerNote");
+
+        return buildAuton(
+            new Pose2d(startToPiece1.getInitialPose().getTranslation(), new Rotation2d()),
+            shoot(),
+            goIntake(startToPiece1),
+            shoot()
+        );
+    }
+
+    /** Starts source side. Shoots preloaded note, intakes bottom note, shoots note. */
+    public SequentialCommandGroup getBottomTwoPiece() {
+        
+        ChoreoTrajectory startToPiece1 = Choreo.getTrajectory("B1-BottomStartToBottomNote");
+
+        return buildAuton(
+            new Pose2d(startToPiece1.getInitialPose().getTranslation(), new Rotation2d()),
+            shoot(),
+            goIntake(startToPiece1),
+            shoot()
+        );
+    }
+
+    /**
+     * Starts: right in front of subwoofer. Shoots preloaded note, intakes middle note, shoots,
+     * intakes top note, shoots.
+     */
+    public SequentialCommandGroup getMiddleThreePiece() {
```

Comment:
These being in the same file is interesting 🤔

Reply:
What you could do is have `buildAuton` be public and separate files which are basically invocations to `buildAuton`? Then you keep this auton builder abstraction but still have separation of separate logic.

Kinda like the base auton class last year but better, perhaps?